views/js/runner/plugins/content/responsiveness/collapser.js

```diff
@@ -108,29 +211,78 @@ define([
                 }
             }
 
-            function shouldCollapseInOrder() {
-                return config.collapseInOrder && _.isArray(config.collapseOrder) && config.collapseOrder.length;
-            }
-
             function collapseInOrder() {
-                var collapseOrderCopy = _.clone(config.collapseOrder),
+                var collapsiblesCopy = _.clone(allCollapsibles),
                     toCollapse;
 
-                while (collapseNeeded() && collapseOrderCopy.length) {
-                    toCollapse = collapseOrderCopy.shift();
-                    $actionsBar.find(toCollapse).toggleClass(collapseCls, true);
+                while (collapseNeeded() && collapsiblesCopy.length) {
+                    toCollapse = collapsiblesCopy.shift();
+                    toCollapse.$elements.addClass(collapseCls);
                 }
             }
 
-            testRunner
-                .on('renderitem loaditem', function() {
-                    collapseAll(false);
-                    if (shouldCollapseInOrder()) {
-                        collapseInOrder();
-                    } else {
-                        collapseAll(collapseNeeded());
+            function collapseNeeded() {
+                return getToolbarWidth() > getAvailableWidth();
+            }
+
+
+            function expandAll() {
+                if (expandPossible(totalExtraWidth)) {
+                    allCollapsibles.forEach(function(collapsible) {
+                        collapsible.$elements.removeClass(collapseCls);
+                    });
+                }
+            }
+
+            function expandInOrder() {
+                var collapsiblesCopy = _.clone(allCollapsibles),
+                    shouldStop = false;
+
+                collapsiblesCopy.reverse();
+
+                collapsiblesCopy.forEach(function (toExpand) {
+                    if (toExpand.$elements.hasClass(collapseCls) && ! shouldStop) {
+                        if (expandPossible(toExpand.extraWidth)) {
+                            toExpand.$elements.removeClass(collapseCls);
+                        } else {
+                            shouldStop = true;
```

Comment:
Rather using such complex structure, you could also use `_.forEachRight()`, that loops from the end to the beginning, and then return `false` to break the loop when needed.  That way no need to clone the array and to manage a stop flag...

https://lodash.com/docs/2.4.2#forEachRight